python-avd/pyavd/_cv/workflows/models.py

```diff
@@ -73,6 +73,131 @@ class CVPathfinderMetadata:
     device: CVDevice | None = None
 
 
+@dataclass
+class CVWorkspaceBuildConfigValidationError:
+    error_code: (
+        Literal[
+            "UNSPECIFIED",
+            "DEVICE_WARNING",
+            "DEVICE_ERROR",
+            "UNREACHABLE_DEVICE",
+            "CONFIG_FILTER_ERROR",
+            "INTERNAL",
+        ]
+        | None
+    ) = None
+    """Error code of the returned error or warning."""
+    error_msg: str | None = None
+    """EOS-returned error message."""
+    line_num: int | None = None
+    """Line number of the violating configuration line within the configlet."""
+    configlet_name: str | None = None
+    """Name of the Studio-generated configlet which raised validation error."""
+
+
+@dataclass
+class CVWorkspaceBuildConfigValidationResult:
+    errors: list[CVWorkspaceBuildConfigValidationError] = field(default_factory=list)
+    warnings: list[CVWorkspaceBuildConfigValidationError] = field(default_factory=list)
+
+
+@dataclass
+class CVWorkspaceBuildImageValidationError:
+    sku: str | None = None
+    """Name of the SKU."""
+    error_code: (
+        Literal[
+            "UNSPECIFIED",
+            "SUPPORT_NOT_INTRODUCED",
+            "SUPPORT_REMOVED",
+            "DEVICE_UNREACHABLE",
+            "VALIDATION_FAILED",
+            "GET_PROPOSED_IMAGE_INFO_FAILED",
+            "GET_RUNNING_IMAGE_INFO_FROM_ACTIVE_SUPERVISOR_FAILED",
+            "GET_RUNNING_IMAGE_INFO_FROM_PEER_SUPERVISOR_FAILED",
+            "EOS_TA_ARCHITECTURE_INCOMPATIBLE",
+            "TA_CV_INCOMPATIBLE",
+            "EOS_CV_INCOMPATIBLE",
+            "EOS_SUPPORT_NOT_INTRODUCED",
+            "EOS_SUPPORT_REMOVED",
+            "PHYSICAL_DEVICE_EOS_INCOMPATIBLE",
+            "TA_EMBEDDEDEXT_INCOMPATIBLE",
+            "DEVICE_EOS_2GB_INCOMPATIBLE",
+            "EOS_EXTENSION_VERSION_INCOMPATIBLE",
+            "ARCH_UNSUPPORTED",  # NEW CODE ADDED IN CVP BUT NOT TRACKED BY pyavd/_cv/api/arista/imagestatus/v1/__init__.py
+        ]
+        | None
+    ) = None
+    """Error code of the returned error."""
+    error_msg: str | None = None
+    """EOS-returned error message."""
+
+
+@dataclass
+class CVWorkspaceBuildImageValidationWarning:
+    sku: str | None = None
+    """Name of the sku."""
+    warning_code: (
```

Comment:
Try to grab these from the generated api code so we don't have to maintain it here. Otherwise just use a string.

Reply:
Fixed